src/chain.h

```diff
@@ -275,6 +276,11 @@ class CBlockIndex
      */
     bool HaveTxsDownloaded() const { return nChainTx != 0; }
 
+    NodeSeconds Time() const
```

Comment:
nit: This will end up renaming `GetBlockTime()`, right? (assuming that in the long run `GetBlockTime()` would be entirely replaced by `Time()`)

we could avoid that with something like the following:
```c++
    template <typename T = int64_t>
    auto GetBlockTime() const -> T
    {
        static_assert(std::is_same<T, int64_t>::value || std::is_same<T, NodeSeconds>::value);

        if constexpr (std::is_same<T, int64_t>::value) {
            return (T)nTime;
        } else {
            return NodeSeconds{std::chrono::seconds{nTime}};
        }
    }
```

Reply:
I picked a different name because it felt weird to repeat the class type in the method again. That is, calling `block.GetBlockTime()` seems better replaced by `block.GetTime()` or `block.Time()`.